### PR TITLE
Handle empty values for better analytics

### PR DESCRIPTION
**Pull request by fditrapani**
Shortly after launching our new Site segmenting screen, I noticed it was hard to create Tracks funnels for the following cases:
- If someone didn't enter a site title,
- If someone didn't enter a site topic,
- Seeing the combination of goals people picked.

This PR makes some updates that will allow us to build better funnels for those use cases. 

## Testing
* Turn on debugging by adding `localStorage.setItem( 'debug', 'calypso:analytics*' );` to the console.
* Refresh your screen.
* Visit `/start/.
* Don't enter any values in `What would you like to name your site?` or `What will your site be about?`.
* Select `Showcase your portfolio`, and `Sell products or collect payments` from primary goals in that order.
* Click next and check the console. 
* `field: "Site title"` and `field: "Site topic"` should have a value of N/A. and `field: "Site goal selections"` should equal `"sell,showcase"`.

cc @taggon

### CODE REVIEW

File: client/signup/steps/about/index.jsx [hunk 1]
```diff
@@ -254,23 +254,25 @@ class AboutStep extends Component {
 		const siteTitleInput = formState.getFieldValue( this.state.form, 'siteTitle' );
 		const siteGoalsInput = formState.getFieldValue( this.state.form, 'siteGoals' );
 		const siteGoalsArray = siteGoalsInput.split( ',' );
+		const siteGoalsGroup = siteGoalsArray.sort().join();
```
Comment (fditrapani) [inline client/signup/steps/about/index.jsx]: this ensures that the order is always the same.

Comment (dmsnell) [reply client/signup/steps/about/index.jsx]: is there any reason for two variables?

```js
const siteGoalsArray = siteGoalsInput.split( ',' ).sort().join();
```

File: client/signup/steps/about/index.jsx [hunk 2]
```diff
@@ -254,23 +254,25 @@ class AboutStep extends Component {
...
 			this.props.setSiteTitle( siteTitleValue );
-			this.props.recordTracksEvent( 'calypso_signup_actions_user_input', {
-				field: 'Site title',
-				value: siteTitleInput,
-			} );
 		}
 
+		this.props.recordTracksEvent( 'calypso_signup_actions_user_input', {
+			field: 'Site title',
+			value: siteTitleInput !== '' ? siteTitleInput : 'N/A',
```
Comment (dmsnell) [inline client/signup/steps/about/index.jsx]: not important, but we should also be able to shorten to `value: siteTitleInput || 'N/A'`

**Comment by fditrapani** [comment]
Thanks for the suggestions @dmsnell. I'd be happy to incorporate them for nicer more succinct code :)

**Pull request merged**